marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -482,8 +482,23 @@ function handleDuplicateSources (propToValidate, arrayWithDuplicates) {
   return deDupedArray;
 }
 
+function addDocumentMetadataToSearchResults(searchResponse) {
+  searchResponse.results.forEach(result => {
+    let metadata = {};
+    let documentMetadata = xdmp.documentGetMetadata(result.uri);
+    if(documentMetadata) {
+      metadata["Flow"] = documentMetadata.datahubCreatedInFlow ? documentMetadata.datahubCreatedInFlow : "";
+      metadata["Step"] = documentMetadata.datahubCreatedByStep ? documentMetadata.datahubCreatedByStep : "";
+      metadata["createdOn"] = documentMetadata.datahubCreatedOn ? documentMetadata.datahubCreatedOn : "";
+      metadata["sources"] = getEntitySources(result.uri);
+    }
+    result["metadata"] = metadata;
```

Comment:
To make it clear that this is a hub-specific addition to the search result, let's call this "hubMetadata". Otherwise, a developer could see this and think it's provided by the Search API, and specifically that it's the metadata keys on a document.